src/backend/common/serializer.h

```diff
@@ -512,7 +512,7 @@ class CopySerializeOutput : public SerializeOutput {
  public:
   // Start with something sizeable so we avoid a ton of initial
   // allocations.
-  static const int INITIAL_SIZE = 8388608;
+  static const int INITIAL_SIZE = 1024*4; // the previous value is 8388608 which is too big for network
```

Comment:
You could add another constructor that takes in the `initial_size` as a parameter and use it in the networking stuff, and leave the default `8388608` as it is.